Fix TypeScript compilation errors in test files

## Problem

Running `npx tsc --noEmit` reveals 21 TypeScript compilation errors across 4 test files. The project currently has `"strict": true` enabled in tsconfig.json, which is catching type safety issues in test code.

## Error Summary

**21 total errors across these files:**
- `src/blockchain/adapters/__tests__/BattleNadsAdapter.test.ts` (1 error)
- `src/hooks/game/__tests__/useCachedDataFeed.test.tsx` (14 errors)
- `src/mappers/__tests__/mappers.test.ts` (2 errors)
- `src/utils/__tests__/areaId.test.ts` (4 errors)

## Error Categories

### 1. Jest Mock Type Errors (12 errors)
```
Property 'mockReturnValue' does not exist on type '...'
Property 'mockResolvedValue' does not exist on type '...'
Property 'mockRejectedValue' does not exist on type '...'
```
**Root cause**: Jest mock methods not properly typed for the objects being mocked.

### 2. Strict Null Checks (3 errors)
```
'result' is possibly 'null'
'result.sessionKeyData' is possibly 'null'
```
**Root cause**: Missing null/undefined checks in test assertions.

### 3. BigInt Type Mismatches (5 errors)
```
Argument of type 'string' is not assignable to parameter of type 'bigint'
Argument of type 'number' is not assignable to parameter of type 'bigint'
Argument of type 'null' is not assignable to parameter of type 'bigint'
```
**Root cause**: Functions expecting `bigint` but tests passing other types.

### 4. Database Mock Issues (1 error)
```
Object literal may only specify known properties, and 'isValid' does not exist in type 'SessionKeyData'
```
**Root cause**: Mock data doesn't match expected interface.

## Current TypeScript Configuration

The project uses strict TypeScript settings:
```json
{
  "compilerOptions": {
    "strict": true,
    "target": "ES2020",
    "types": ["node", "jest", "@testing-library/jest-dom"]
  }
}
```

## Proposed Solutions

We need to decide on approach:

### Option A: Update tsconfig.json
- Add more lenient settings for test files
- Configure better Jest type support
- Add `@types/jest` with proper augmentations

### Option B: Fix Individual Errors
- Add proper type assertions in tests
- Fix null checks with proper guards
- Convert BigInt parameters correctly
- Update mock interfaces to match expectations

### Option C: Hybrid Approach
- Update tsconfig for legitimate configuration gaps
- Fix actual type safety issues in test code

## Full Error Output

<details>
<summary>Complete TypeScript Error List</summary>

```
src/blockchain/adapters/__tests__/BattleNadsAdapter.test.ts(114,9): error TS2353: Object literal may only specify known properties, and 'isValid' does not exist in type 'SessionKeyData'.
src/hooks/game/__tests__/useCachedDataFeed.test.tsx(71,25): error TS2339: Property 'mockReturnValue' does not exist on type '{ (index: string | string[]): WhereClause<StoredEvent, string, InsertType<StoredEvent, "owner">>; (equalityCriterias: { [key: string]: any; }): Collection<...>; }'.
src/hooks/game/__tests__/useCachedDataFeed.test.tsx(72,31): error TS2339: Property 'mockReturnValue' does not exist on type '{ (index: string | string[]): WhereClause<StoredChatMessage, string, InsertType<StoredChatMessage, "owner">>; (equalityCriterias: { ...; }): Collection<...>; }'.
src/hooks/game/__tests__/useCachedDataFeed.test.tsx(73,29): error TS2339: Property 'mockReturnValue' does not exist on type '{ (index: string | string[]): WhereClause<StoredDataBlock, string, InsertType<StoredDataBlock, "owner">>; (equalityCriterias: { ...; }): Collection<...>; }'.
src/hooks/game/__tests__/useCachedDataFeed.test.tsx(74,27): error TS2339: Property 'mockResolvedValue' does not exist on type '(item: InsertType<StoredCharacterMetadata, "owner">, key?: string | undefined) => PromiseExtended<string>'.
src/hooks/game/__tests__/useCachedDataFeed.test.tsx(75,43): error TS2345: Argument of type '(mode: TransactionMode, tables: string | Table<any, any, any>, callback: string | Table<any, any, any>) => Promise<any>' is not assignable to parameter of type '(mode: TransactionMode, table: string | Table<any, any, any>, table2: string | Table<any, any, any>, table3: string | Table<any, any, any>, table5: string | Table<...>, scope: (trans: Transaction & { ...; }) => unknown) => PromiseExtended<...>'.
src/hooks/game/__tests__/useCachedDataFeed.test.tsx(76,20): error TS2349: This expression is not callable.
src/hooks/game/__tests__/useCachedDataFeed.test.tsx(137,29): error TS2339: Property 'mockReturnValue' does not exist on type '{ (index: string | string[]): WhereClause<StoredDataBlock, string, InsertType<StoredDataBlock, "owner">>; (equalityCriterias: { ...; }): Collection<...>; }'.
src/hooks/game/__tests__/useCachedDataFeed.test.tsx(168,29): error TS2339: Property 'mockReturnValue' does not exist on type '{ (index: string | string[]): WhereClause<StoredDataBlock, string, InsertType<StoredDataBlock, "owner">>; (equalityCriterias: { ...; }): Collection<...>; }'.
src/hooks/game/__tests__/useCachedDataFeed.test.tsx(219,23): error TS2339: Property 'mockResolvedValue' does not exist on type '(item: InsertType<StoredEvent, "owner">, key?: string | undefined) => PromiseExtended<string>'.
src/hooks/game/__tests__/useCachedDataFeed.test.tsx(220,29): error TS2339: Property 'mockResolvedValue' does not exist on type '(item: InsertType<StoredChatMessage, "owner">, key?: string | undefined) => PromiseExtended<string>'.
src/hooks/game/__tests__/useCachedDataFeed.test.tsx(228,25): error TS2339: Property 'mockReturnValue' does not exist on type '{ (index: string | string[]): WhereClause<StoredEvent, string, InsertType<StoredEvent, "owner">>; (equalityCriterias: { [key: string]: any; }): Collection<...>; }'.
src/hooks/game/__tests__/useCachedDataFeed.test.tsx(229,31): error TS2339: Property 'mockReturnValue' does not exist on type '{ (index: string | string[]): WhereClause<StoredChatMessage, string, InsertType<StoredChatMessage, "owner">>; (equalityCriterias: { ...; }): Collection<...>; }'.
src/hooks/game/__tests__/useCachedDataFeed.test.tsx(231,43): error TS2345: Argument of type '(mode: TransactionMode, tables: string | Table<any, any, any>, callback: string | Table<any, any, any>) => Promise<any>' is not assignable to parameter of type '(mode: TransactionMode, table: string | Table<any, any, any>, table2: string | Table<any, any, any>, table3: string | Table<any, any, any>, table5: string | Table<...>, scope: (trans: Transaction & { ...; }) => unknown) => PromiseExtended<...>'.
src/hooks/game/__tests__/useCachedDataFeed.test.tsx(232,20): error TS2349: This expression is not callable.
src/hooks/game/__tests__/useCachedDataFeed.test.tsx(272,23): error TS2339: Property 'mockRejectedValue' does not exist on type '(item: InsertType<StoredEvent, "owner">, key?: string | undefined) => PromiseExtended<string>'.
src/mappers/__tests__/mappers.test.ts(37,14): error TS18047: 'result' is possibly 'null'.
src/mappers/__tests__/mappers.test.ts(38,14): error TS18047: 'result' is possibly 'null'.
src/mappers/__tests__/mappers.test.ts(38,14): error TS18047: 'result.sessionKeyData' is possibly 'null'.
src/utils/__tests__/areaId.test.ts(71,28): error TS2345: Argument of type 'string' is not assignable to parameter of type 'bigint'.
src/utils/__tests__/areaId.test.ts(72,28): error TS2345: Argument of type 'string' is not assignable to parameter of type 'bigint'.
src/utils/__tests__/areaId.test.ts(73,28): error TS2345: Argument of type 'number' is not assignable to parameter of type 'bigint'.
src/utils/__tests__/areaId.test.ts(75,28): error TS2345: Argument of type 'null' is not assignable to parameter of type 'bigint'.
src/utils/__tests__/areaId.test.ts(76,28): error TS2345: Argument of type 'undefined' is not assignable to parameter of type 'bigint'.
```

</details>

## Impact

- **Production code**: Not affected - all errors are in test files
- **CI/CD**: TypeScript compilation checks may be failing
- **Developer experience**: `tsc` errors create noise during development
- **Type safety**: Missing proper type coverage in test scenarios

## Acceptance Criteria

- [ ] All TypeScript compilation errors resolved
- [ ] Tests continue to pass with proper typing
- [ ] Decision documented on approach taken (config vs code fixes)
- [ ] No regression in type safety for production code

--- a/src/blockchain/adapters/__tests__/BattleNadsAdapter.test.ts
+++ b/src/blockchain/adapters/__tests__/BattleNadsAdapter.test.ts
@@ -109,10 +109,13 @@ describe('BattleNadsAdapter', () => {
     
     it('returns correct data structure', async () => {
       const mockReturn: ContractTypes.SessionKeyData = {
+        owner: '0xOwner',
         key: '0xSessionKey',
+        balance: 1000n,
+        targetBalance: 2000n,
+        ownerCommittedAmount: 500n,
+        ownerCommittedShares: 100n,
         expiration: 9999999n,
-        isValid: true,
-        // No 'balance' property expected here
       };
       mockContractMethods.getCurrentSessionKeyData.mockResolvedValueOnce(mockReturn);
       

--- a/src/hooks/game/__tests__/useCachedDataFeed.test.tsx
+++ b/src/hooks/game/__tests__/useCachedDataFeed.test.tsx
@@ -68,11 +68,11 @@ describe('useCachedDataFeed', () => {
     };
     
     // Mock all database tables
-    mockDb.events.where.mockReturnValue(mockWhere as any);
-    mockDb.chatMessages.where.mockReturnValue(mockWhere as any);
-    mockDb.dataBlocks.where.mockReturnValue(mockWhere as any);
-    mockDb.characters.put.mockResolvedValue(undefined as any);
-    mockDb.transaction.mockImplementation(async (mode, tables, callback) => {
+    (mockDb.events.where as jest.Mock).mockReturnValue(mockWhere as any);
+    (mockDb.chatMessages.where as jest.Mock).mockReturnValue(mockWhere as any);
+    (mockDb.dataBlocks.where as jest.Mock).mockReturnValue(mockWhere as any);
+    (mockDb.characters.put as jest.Mock).mockResolvedValue(undefined as any);
+    (mockDb.transaction as jest.Mock).mockImplementation(async (mode: any, tables: any, callback: any) => {
       return await callback();
     });
   });
@@ -134,7 +134,7 @@ describe('useCachedDataFeed', () => {
       }),
     };
     
-    mockDb.dataBlocks.where.mockReturnValue(mockWhere as any);
+    (mockDb.dataBlocks.where as jest.Mock).mockReturnValue(mockWhere as any);
 
     const { result } = renderHook(() => useCachedDataFeed(mockOwner, mockCharacterId), {
       wrapper: createWrapper(),
@@ -165,7 +165,7 @@ describe('useCachedDataFeed', () => {
       }),
     };
     
-    mockDb.dataBlocks.where.mockReturnValue(mockWhere as any);
+    (mockDb.dataBlocks.where as jest.Mock).mockReturnValue(mockWhere as any);
 
     const consoleSpy = jest.spyOn(console, 'error').mockImplementation();
 
@@ -216,8 +216,8 @@ describe('storeEventData', () => {
 
   beforeEach(() => {
     jest.clearAllMocks();
-    mockDb.events.add.mockResolvedValue(undefined as any);
-    mockDb.chatMessages.add.mockResolvedValue(undefined as any);
+    (mockDb.events.add as jest.Mock).mockResolvedValue(undefined as any);
+    (mockDb.chatMessages.add as jest.Mock).mockResolvedValue(undefined as any);
     
     // Mock first() to return null (no existing events)
     const mockWhere = {
@@ -225,10 +225,10 @@ describe('storeEventData', () => {
         first: jest.fn().mockResolvedValue(null),
       }),
     };
-    mockDb.events.where.mockReturnValue(mockWhere as any);
-    mockDb.chatMessages.where.mockReturnValue(mockWhere as any);
+    (mockDb.events.where as jest.Mock).mockReturnValue(mockWhere as any);
+    (mockDb.chatMessages.where as jest.Mock).mockReturnValue(mockWhere as any);
     
-    mockDb.transaction.mockImplementation(async (mode, tables, callback) => {
+    (mockDb.transaction as jest.Mock).mockImplementation(async (mode: any, tables: any, callback: any) => {
       return await callback();
     });
   });
@@ -269,7 +269,7 @@ describe('storeEventData', () => {
   });
 
   it('should handle database storage errors', async () => {
-    mockDb.events.add.mockRejectedValue(new Error('Storage error'));
+    (mockDb.events.add as jest.Mock).mockRejectedValue(new Error('Storage error'));
     
     const consoleSpy = jest.spyOn(console, 'error').mockImplementation();
 

--- a/src/mappers/__tests__/mappers.test.ts
+++ b/src/mappers/__tests__/mappers.test.ts
@@ -34,8 +34,9 @@ describe('Data Mappers', () => {
       const result = contractToWorldSnapshot(mockPollData as any, '0xOwnerAddress');
       
       // Basic assertions that focus on functionality
-      expect(result.characterID).toBe(mockPollData.characterID);
-      expect(result.sessionKeyData.key).toBe(mockPollData.sessionKeyData.key);
+      expect(result).toBeDefined();
+      expect(result?.characterID).toBe(mockPollData.characterID);
+      expect(result?.sessionKeyData?.key).toBe(mockPollData.sessionKeyData.key);
     });
   });
 }); 

--- a/src/utils/__tests__/areaId.test.ts
+++ b/src/utils/__tests__/areaId.test.ts
@@ -68,12 +68,12 @@ describe('Area ID utilities', () => {
     });
 
     it('should reject invalid area IDs', () => {
-      expect(isValidAreaID('')).toBe(false);
-      expect(isValidAreaID('invalid')).toBe(false);
-      expect(isValidAreaID(123)).toBe(false); // number instead of bigint
+      expect(isValidAreaID('' as any)).toBe(false);
+      expect(isValidAreaID('invalid' as any)).toBe(false);
+      expect(isValidAreaID(123 as any)).toBe(false); // number instead of bigint
       expect(isValidAreaID(-1n)).toBe(false); // negative bigint
-      expect(isValidAreaID(null)).toBe(false);
-      expect(isValidAreaID(undefined)).toBe(false);
+      expect(isValidAreaID(null as any)).toBe(false);
+      expect(isValidAreaID(undefined as any)).toBe(false);
     });
   });
 });

